components/butterfly/src/server/outbound.rs

```diff
@@ -302,11 +300,8 @@ pub fn populate_membership_rumors_mlr(server: &Server,
         }
     }
 
-    // NOTE: the way this is currently implemented, this is grabbing
-    // the 5 coolest (but still warm!) Member rumors.
     let rumors: Vec<RumorKey> = server
-        .rumor_heat
-        .currently_hot_rumors(&target.id)
+        .keys_for_live_rumors()
```

Comment:
I'll confess that I missed the `take(5)` at the end.  In my head, this was going to send all of the live rumors every time, which, while sub-optimal, at least gets us completeness guarantees.  I agree that this is something we should re-think.

Reply:
For now, I'll update this to send all the members, which is likely not the best option, but probably better than sending the first 5 every time.  It'd be worth having a larger discussion about this.

Reply:
I ended up changing this to selecting 5 members at random, since that's closer to the intent of the SWIM algorithm, IIRC.

Reply:
I'm still open to discussing it further, though.

Reply:
After reading through the SWIM paper again, particularly Section 4.1 on Infection-Style Dissemination Component, I think we want to retain some of the prior "heat" behavior (but don't call it "heat", since we agree that what we're doing here is not the "heat" of the Xerox gossip paper).

That section talks about implementation details for how to select members to add to the piggyback component. It talks about maintaining a buffer of recent membership updates, along with a local count for each buffer element. The count is how many times that item has been piggybacked by the node. If you've got more information to piggyback than room to piggyback, you prefer to send out items that you haven't already sent out as much.

That sounds like what this bit of code was attempting to do, but it was conflating the needs of the SWIM component with the other non-SWIM rumors by pushing that into the "rumor heat" mechanism.

I think following the guidance of the SWIM paper here, as well as making an effort to more cleanly separate our SWIM from our non-SWIM concerns, would be useful here. So, I think that removing the existing "rumor heat" altogether is overall a good call, though I need to read further to understand how it impacts the non-SWIM data.

(Also, the `take(5)` portion of this seems to be geared around keeping the overall size of the data contained in a UDP packet (based on the SWIM paper, at least). Whether that assumption is valid here, or whether 5 actually completely fits, I'm not sure. That's the tricky thing about magic numbers, I suppose.)

Reply:
This is a great idea.  I'll give it a go.

Reply:
Well said, @christophermaier. I think at some point, we'll need to evaluate whether the piggybacking is really piggybacking at all. If we add too much membership data and end up sending twice or more as many packets, that's going to undermine our purposes here.

Off the top of my head, I'm not sure how we'd do that, and I think we can punt for now, but it's important to revisit.